dynosaur_derive/src/lib.rs

```diff
@@ -53,19 +53,28 @@ pub fn dynosaur(
     let attrs = parse_macro_input!(attr as Attrs);
     let item_trait = parse_macro_input!(item as ItemTrait);
 
+    let struct_ident = &attrs.ident;
     let expanded_trait_to_dyn = expand_trait_async_fns_to_dyn(&item_trait);
     let erased_trait = mk_erased_trait(&expanded_trait_to_dyn);
     let erased_trait_blanket_impl = mk_erased_trait_blanket_impl(&item_trait.ident, &erased_trait);
     let dyn_struct = mk_dyn_struct(&attrs.ident, &erased_trait);
-    let dyn_struct_impl_item = mk_dyn_struct_impl_item(&attrs.ident, &item_trait);
+    let dyn_struct_impl_item = mk_dyn_struct_impl_item(struct_ident, &item_trait);
+    let struct_inherent_impl =
+        mk_struct_inherent_impl(&attrs.ident, &item_trait.ident, &erased_trait);
 
     quote! {
         #item_trait
 
-        #erased_trait
-        #erased_trait_blanket_impl
-        #dyn_struct
-        #dyn_struct_impl_item
+        mod _dynosaur_macro {
```

Comment:
I think this will need to include the trait name or something, so multiple derives in the same file don't conflict.

Reply:
Uhh yes, very silly mistake. Fixed.